conda/exceptions.py

```diff
@@ -590,36 +593,181 @@ def print_conda_exception(exception):
+        do_upload = not context.offline and context.always_yes
+    elif not isatty:
+        ask_for_upload = False
+        do_upload = not context.offline and context.always_yes
+    else:
+        ask_for_upload = True
+        do_upload = False
 
-    stderrlogger = getLogger('stderr')
+    return ask_for_upload, do_upload
 
-    from .base.context import context
+
+def _print_exception_message_and_prompt(context, error_report):
+    ask_for_upload, do_upload = _calculate_ask_do_upload(context)
+
+    stdin = None
     if context.json:
         from .cli.common import stdout_json
-        stdout_json(dict(error=traceback))
+        stdout_json(error_report)
     else:
-        message = """\
-An unexpected error has occurred.
-Please consider posting the following information to the
-conda GitHub issue tracker at:
+        message_builder = []
+        if not ask_for_upload:
+            message_builder.append(
+                "An unexpected error has occurred. Conda has prepared the following report."
+            )
+        message_builder.append('')
+        message_builder.append('`$ %s`' % error_report['command'])
+        message_builder.append('')
+        message_builder.extend('    ' + line for line in error_report['traceback'].splitlines())
+        message_builder.append('')
+        if error_report['conda_info']:
+            from .cli.main_info import get_main_info_str
+            try:
+                message_builder.append(get_main_info_str(error_report['conda_info']))
+            except Exception as e:
+                message_builder.append('conda info could not be constructed.')
+                message_builder.append('%r' % e)
+        message_builder.append('')
+
+        if ask_for_upload:
+            message_builder.append(
+                "An unexpected error has occurred. Conda has prepared the above report."
+            )
+            message_builder.append(
+                "Would you like conda to send this report to the core maintainers?"
+            )
+            message_builder.append(
+                "[y/N]: "
+            )
+        sys.stderr.write('\n'.join(message_builder))
+        if ask_for_upload:
+            try:
+                stdin = timeout(40, input)
+                do_upload = stdin and boolify(stdin)
+
+            except Exception as e:  # pragma: no cover
+                log.debug('%r', e)
+                do_upload = False
+
+    return do_upload, ask_for_upload, stdin
+
+
+def _execute_upload(context, error_report):
+    headers = {
+        'User-Agent': context.user_agent,
+    }
+    _timeout = context.remote_connect_timeout_secs, context.remote_read_timeout_secs
+    data = json.dumps(error_report, sort_keys=True, cls=EntityEncoder) + '\n'
+    response = None
+    try:
+        # requests does not follow HTTP standards for redirects of non-GET methods
+        # That is, when following a 301 or 302, it turns a POST into a GET.
+        # And no way to disable.  WTF
```

Comment:
@goanpeca When you implement this in Navi, definitely be aware of the above comment and the hack below.  It's an evil evil requests behavior.